integrations/GA4/browser.js

```diff
@@ -0,0 +1,211 @@
+    const item = {};
+    Object.keys(props).forEach((key) => {
+      destParameter.forEach((param) => {
+        if (key === param.src) {
+          if (Array.isArray(param.dest)) {
+            param.dest.forEach((d) => {
+              const result = d.split(".");
+              // Here we only support mapping single level object mapping.
+              // To Do Future Scope :: implement using recursion to handle multi level prop mapping
+              if (result.length > 1) {
+                const levelOne = result[0];
+                const levelTwo = result[1];
+                item[levelTwo] = props[key];
+                if (!destinationProperties[levelOne]) {
+                  destinationProperties[levelOne] = [];
+                  destinationProperties[levelOne].push(item);
+                }
+              } else {
+                destinationProperties[result] = props[key];
+              }
+            });
+          } else {
+            destinationProperties[param.dest] = props[key];
+          }
+        }
+      });
+    });
+    return destinationProperties;
+  }
+
+  getDestinationItemProperties(products, item) {
+    const items = [];
+    let obj = {};
+    products.forEach((p) => {
+      obj = {
+        ...this.getDestinationEventProperties(p, itemParameter),
+        ...(item && item[0]),
+      };
+      items.push(obj);
+    });
+    return items;
+  }
+
+  track(rudderElement) {
+    let { event } = rudderElement.message;
+    const { properties } = rudderElement.message;
+    const { products } = properties;
+    let destinationProperties = {};
+    if (!event || this.isReservedName(event)) {
+      throw Error("Cannot call un-named/reserved named track event");
+    }
+
+    const obj = this.getDestinationEvent(event);
+    if (obj) {
+      if (products && Array.isArray(products)) {
+        event = obj.dest;
+        // eslint-disable-next-line no-const-assign
+        destinationProperties = this.getDestinationEventProperties(
+          properties,
+          eventParameter
+        );
+        destinationProperties.items = this.getDestinationItemProperties(
+          products,
+          destinationProperties.items
+        );
+      } else {
+        event = obj.dest;
+        if (!obj.hasItem) {
+          // eslint-disable-next-line no-const-assign
+          destinationProperties = this.getDestinationEventProperties(
+            properties,
+            eventParameter
+          );
+        } else {
+          // create items
+          destinationProperties.items = this.getDestinationItemProperties([
+            properties,
+          ]);
+        }
+      }
+    } else {
+      destinationProperties = properties;
+    }
+
+    this.sendGAEvent(event, destinationProperties);
+  }
+
+  identify(rudderElement) {
+    if (this.sendUserId && rudderElement.message.userId) {
+      const userId = this.analytics.userId || this.analytics.anonymousId;
+      window.gtag("config", this.measurementId, {
```

Comment:
this works even if identify is not the first call?

Reply:
Yes, user is identified even if no identify call is made. As soon as ga scripts load he gets identified in google analytics.